staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -0,0 +1,408 @@
+	args = args[1:]
+	cmd, _, err := rootCmd.Find(args)
+	if err != nil {
+		fmt.Fprintf(errOut, "Warning: command not found %v\n", err)
+		return nil
+	}
+
+	for _, c := range kuberc.Overrides {
+		parsedCmds := strings.Fields(c.Command)
+		overrideCmd, _, err := rootCmd.Find(parsedCmds)
+		if err != nil {
+			fmt.Fprintf(errOut, "Warning: command %q not found to set kuberc override\n", c.Command)
+			continue
+		}
+		if overrideCmd.Name() != cmd.Name() {
+			continue
+		}
+
+		if _, ok := p.aliases[cmd.Name()]; ok {
+			return fmt.Errorf("alias %s can not be overridden", cmd.Name())
+		}
+
+		// This function triggers merging the persistent flags in the parent commands.
+		_ = cmd.InheritedFlags()
+
+		for _, fl := range c.Flags {
+			existingFlag := cmd.Flag(fl.Name)
+			if existingFlag == nil {
+				return fmt.Errorf("invalid flag %s for command %s", fl.Name, c.Command)
+			}
+			if searchInArgs(existingFlag.Name, existingFlag.Shorthand, cmd.Flags(), args) {
+				// Don't modify the value implicitly, if it is passed in args explicitly
+				continue
+			}
+			err = cmd.Flags().Set(fl.Name, fl.Default)
+			if err != nil {
+				return fmt.Errorf("could not apply override value %s to flag %s in command %s err: %w", fl.Default, fl.Name, c.Command, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// applyAliases firstly appends all defined aliases in kuberc file to the root command.
+// Since there may be several alias definitions belonging to the same command, it extracts the
+// alias that is currently executed from args. After that it sets the flag definitions in alias as default values
+// of the command. Lastly, others parameters (e.g. resources, etc.) that are passed as arguments in kuberc
+// is appended into the command args.
+func (p *Preferences) applyAliases(rootCmd *cobra.Command, kuberc *config.Preference, args []string, errOut io.Writer) ([]string, error) {
+	_, _, err := rootCmd.Find(args[1:])
+	if err == nil {
+		// Command is found, no need to continue for aliasing
+		return args, nil
+	}
+
+	var aliasArgs *aliasing
+
+	var commandName string // first "non-flag" arguments
+	for _, arg := range args[1:] {
+		if !strings.HasPrefix(arg, "-") {
+			commandName = arg
+			break
+		}
+	}
+	commandName = strings.TrimSpace(commandName)
+
+	for _, alias := range kuberc.Aliases {
+		p.aliases[alias.Name] = struct{}{}
+		if alias.Name != commandName {
+			continue
+		}
+
+		// do not allow shadowing built-ins
+		if _, _, err := rootCmd.Find([]string{alias.Name}); err == nil {
+			fmt.Fprintf(errOut, "Warning: Setting alias %q to a built-in command is not supported\n", alias.Name)
+			break
+		}
+
+		commands := strings.Fields(alias.Command)
+		existingCmd, flags, err := rootCmd.Find(commands)
+		if err != nil {
+			return args, fmt.Errorf("command %q not found to set alias %q: %v", alias.Command, alias.Name, flags)
+		}
+
+		newCmd := *existingCmd
+		newCmd.Use = alias.Name
```

Comment:
does overriding `Use` mess with the help that gets displayed? what would happen if they ran `kubectl myalias --help`?

Reply:
This is an example of alias;

```yaml
apiVersion: kubectl.config.k8s.io/v1alpha1
kind: Preference
aliases:
- name: crns
  command: create namespace
  args:
   - test-kuberc-ns
```

Output of alias --help;

```
$ ./_output/bin/kubectl crns --help
Create a namespace with the specified name.

Aliases:
crns, ns

Examples:
  # Create a new namespace named my-namespace
  kubectl create namespace my-namespace
```

Output of built-in command's help;

```
$ ./_output/bin/kubectl create namespace --help
Create a namespace with the specified name.

Aliases:
namespace, ns

Examples:
  # Create a new namespace named my-namespace
  kubectl create namespace my-namespace
```

Alias name is populated in alias field of actual command.

Reply:
```
$ ./_output/bin/kubectl crns --help
Create a namespace with the specified name.

Aliases:
crns, ns
```

The `ns` there is weird, and wouldn't actually work if you invoked it as `kubectl ns` ... we may need to clear the Aliases from the copied command as well

Reply:
`Alises` field is cleared for alias command.